### PR TITLE
pgsql: add tests with alert metadata, adjust pcap_cnt - v2

### PR DESCRIPTION
**Pull request by jufajardini**
Check for transaction metadata in PGSQL alerts.
Add `engine-analysis` tests for the used rules, as well, to better describe them and compare with expected behavior.

Related to
Task #7000

https://github.com/OISF/suricata-verify/pull/1990 without force-pushes

## Suricata PR

https://github.com/OISF/suricata/pull/11547

## Ticket

If your pull request is related to a Suricata ticket, please provide
the full URL to the ticket here so this pull request can monitor
changes to the ticket status:

Redmine ticket:
https://redmine.openinfosecfoundation.org/issues/7113
https://redmine.openinfosecfoundation.org/issues/7000

### CODE REVIEW

File: tests/pgsql/pgsql-7000-ips/test.yaml [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+    count: 1
+    match:
+      alert.signature: Pgsql table select
+      alert.signature_id: 1
+      app_proto: pgsql
+      direction: to_server
+      event_type: alert
+      payload_printable: ......./...O....user.ju-Test.database.Test.application_name.psql.client_encoding.UTF8..p...6SCRAM-SHA-256.... n,,n=,r=ROtF8e2Fme8+eORLNHTwkZaKp...lc=biws,r=ROtF8e2Fme8+eORLNHTwkZaKtpbEaXYJOnd3qt6QNCsAv0wj,p=I4V0zdtQqrxum6B+QzprHHC0nBD+mVtBWpc+arfXa+c=Q....BEGIN;.Q....DELETE FROM new_table WHERE NAME='Remus';.Q...1DELETE FROM new_table WHERE NAME='Londubat';.Q...1DELETE FROM new_table WHERE NAME='Hermione';.Q.../DELETE FROM new_table WHERE NAME='Maugre';.Q....COMMIT;.Q....BEGIN;.Q...HINSERT INTO new_table VALUES('Hermione', 'prof_gramger@gmail.com');.Q...CINSERT INTO new_table VALUES('Remus', 'prof_lupin@gmail.com');.Q....SELECT 1/0;.Q...EINSERT INTO new_table VALUES('Maugre', 'prof_folloy@gmail.com');.Q...IINSERT INTO new_table VALUES('Londubat', 'prof_londubat@gmail.com');.Q....SELECT * FROM new_table;.Q....COMMIT;.Q....BEGIN;.
+      pcap_cnt: 64
+      pgsql.request.simple_query: INSERT INTO new_table VALUES('Hermione', 'prof_gramger@gmail.com');
```
Comment (inashivb) [inline tests/pgsql/pgsql-7000-ips/test.yaml:302]: xD

File: tests/pgsql/pgsql-7000-ips/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Showcase engine behavior with simple pgsql stream rules and alert metadata, in
+IPS mode.
+
+## Behavior
+
+The 2 rules will match on the PGSQL queries for `SELECT` (sid 1) and `DELETE FROM`,
+(sid 2). However, the combination of IPS mode matching on the traffic as soon as
+the engine sees it with the continuously available stream buffer leads to way
+more matches than could be initially expected. The `payload_printable` EVE field
```
Comment (inashivb) [inline tests/pgsql/pgsql-7000-ips/README.md:11]: if there are way more matches, why does the test fail with `expected 1 got 0`.. messages?

File: tests/pgsql/pgsql-7000-ips/test.yaml (hunk 1 above)
Comment (jufajardini) [reply tests/pgsql/pgsql-7000-ips/test.yaml:302]: Can you tell I was listening to HP books when I created those pcaps? xD

File: tests/pgsql/pgsql-7000-ips/README.md (hunk 2 above)
Comment (inashivb) [reply tests/pgsql/pgsql-7000-ips/README.md:11]: Sorry for being unclear.
What I mean is that if there are more matches then why does the sub test to check for 37 alerts not fail with `expected 37 got 100..` or something?

Comment (jufajardini) [reply tests/pgsql/pgsql-7000-ips/README.md:11]: Ah! I decided to do a test that showcases what we currently have. 🤔

**Comment by jufajardini** [comment]
Rebased with https://github.com/OISF/suricata-verify/pull/2004

File: tests/pgsql/pgsql-7000-ips/README.md (hunk 2 above)
Comment (inashivb) [reply tests/pgsql/pgsql-7000-ips/README.md:11]: hmm. I think the tests should always have the expected behavior so the PR corresponding to it shows that it is bringing in the expected behavior.

Even more confused now. If this is showing current behavior, why does anything fail at all?

Comment (jufajardini) [reply tests/pgsql/pgsql-7000-ips/README.md:11]: > hmm. I think the tests should always have the expected behavior so the PR corresponding to it shows that it is bringing in the expected behavior.

Ok, I can update/change them. Then we would have tests failing until we have a bug fix - in which case I'd like to first confirm that there's a bug, so a ticket is created, then I would also probably split this PR to remove the tests that show the bug 🤔 

> Even more confused now. If this is showing current behavior, why does anything fail at all?

Current behavior with the Suricata PR '-'

Comment (inashivb) [reply tests/pgsql/pgsql-7000-ips/README.md:11]: >  then I would also probably split this PR to remove the tests that show the bug 🤔

indeed. that sounds better.
 
> Current behavior with the Suricata PR '-'

ah sorry, misunderstood. 😓

**Pull request closed without merging**